### PR TITLE
feat: add middlewareModule to nuxt and next storefront packages

### PR DESCRIPTION
**Pull request by WojtekTheWebDev**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSDoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Adds `moduleFromEndpoints` to nuxt and next storefront packages.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/storefront/packages/next/__tests__/apps/app-router/tsconfig.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "lib": ["dom", "dom.iterable", "esnext"],
     "allowJs": true,
     "skipLibCheck": true,
-    "strict": true,
+    "strict": false,
```
Comment (jagoral) [inline packages/storefront/packages/next/__tests__/apps/app-router/tsconfig.json:7]: app should work also in strict mode

File: packages/storefront/packages/nuxt/__tests__/app/package.json [hunk 2]
```diff
@@ -10,6 +10,6 @@
   },
   "devDependencies": {
     "nuxt": "3.7.4",
-    "@vue-storefront/sdk": "*"
+    "@vue-storefront/sdk": "1.4.0-rc.3"
```
Comment (jagoral) [inline packages/storefront/packages/nuxt/__tests__/app/package.json:13]: do we need to specify the exact version?

File: packages/storefront/packages/nuxt/__tests__/app/tsconfig.json [hunk 3]
```diff
@@ -1,3 +1,6 @@
 {
-  "extends": "./.nuxt/tsconfig.json"
+  "extends": "./.nuxt/tsconfig.json",
+  "compilerOptions": {
+    "strict": false
```
Comment (jagoral) [inline packages/storefront/packages/nuxt/__tests__/app/tsconfig.json:4]: as above

**Review by jagoral — CHANGES_REQUESTED** [review]
I think we should also update the Getting started for the SDK https://docs.vuestorefront.io/sdk/getting-started

File: packages/storefront/packages/nuxt/__tests__/app/package.json (hunk 2 above)
Comment (WojtekTheWebDev) [reply packages/storefront/packages/nuxt/__tests__/app/package.json:13]: not really just packages are not linked in monorepo and I needed to use it for test purposes.

File: packages/storefront/packages/nuxt/src/runtime/defineSdkConfig.template [hunk 4]
```diff
@@ -57,12 +55,13 @@ export function defineSdkConfig<TConfig>(config: Config<TConfig>) {
...
 
-    const getCookieHeader = () => useRequestHeaders(["cookie"]);
+    const getRequestHeaders = () => useRequestHeaders();
 
     return config({
       buildModule,
+      moduleFromEndpoints,
       middlewareUrl,
-      getCookieHeader
+      getRequestHeaders
```
Comment (jagoral) [inline packages/storefront/packages/nuxt/src/runtime/defineSdkConfig.template:64]: I'm completely fine with this change, just please remember that it's a breaking change, so we will have to bump the major

File: packages/storefront/packages/next/__tests__/apps/app-router/tsconfig.json (hunk 1 above)
Comment (WojtekTheWebDev) [reply packages/storefront/packages/next/__tests__/apps/app-router/tsconfig.json]: Fixed

File: packages/storefront/packages/nuxt/__tests__/app/tsconfig.json (hunk 3 above)
Comment (WojtekTheWebDev) [reply packages/storefront/packages/nuxt/__tests__/app/tsconfig.json]: Fixed

File: packages/storefront/packages/nuxt/src/runtime/defineSdkConfig.template (hunk 4 above)
Comment (WojtekTheWebDev) [reply packages/storefront/packages/nuxt/src/runtime/defineSdkConfig.template:64]: Thanks for notifying, I'm gonna deprecate it then

File: packages/storefront/packages/nuxt/__tests__/app/package.json (hunk 2 above)
Comment (WojtekTheWebDev) [reply packages/storefront/packages/nuxt/__tests__/app/package.json]: it's gonna be resolved after merging https://github.com/vuestorefront/vue-storefront/pull/7052

File: .changeset/pre.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  "mode": "pre",
+  "tag": "rc",
+  "initialVersions": {
+    "@vue-storefront/cli": "4.5.0",
+    "@vue-storefront/middleware": "3.8.1",
+    "@vue-storefront/multistore": "2.0.1",
+    "@vue-storefront/sdk": "1.3.2"
+  },
+  "changesets": ["brave-mirrors-design", "silver-squids-switch"]
+}
```
Comment (jagoral) [inline .changeset/pre.json:11]: the pre flow in changesets couldn't be used, because the storefront packages are released manually

Comment (WojtekTheWebDev) [reply .changeset/pre.json]: done

**Review by jagoral — APPROVED** [review]

File: packages/storefront/packages/next/CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,31 @@
 # Change log
 
+## 1.1.0
+
+- **[ADDED]** Add `middlewareModule` to `createSdk` params.
```
Comment (jagoral) [inline packages/storefront/packages/next/CHANGELOG.md:5]: I think based on Filip's RFC, we could have

```
**[ADDED]** `middlewareModule` to `createSdk` params.
```

instead, so you read the whole line (including the tag) as a sentence

https://github.com/vuestorefront/vue-storefront/blob/main/packages/changesets/reference-changelog.md#minor-changes

File: packages/storefront/packages/nuxt/CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,45 @@
 # Change log
 
+## 3.1.0
+
+- **[ADDED]** Add `middlewareModule` to `createSdk` params.
```
Comment (jagoral) [inline packages/storefront/packages/nuxt/CHANGELOG.md]: we've got `createSdk` in Next.js, in Nuxt we've added the `middlewareModule` to the `defineSdkConfig` callback's params

File: packages/storefront/packages/nuxt/CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,45 @@
...
+- import { UnifiedApiExtension } from "storefront-middleware/types"
++ import { UnifiedEndpoints } from "storefront-middleware/types"
+
+export default defineSdkConfig(
+-  ({ buildModule, middlewareUrl, getRequestHeaders }) => ({
+-    commerce: buildModule(unifiedModule<UnifiedApiExtension>, {
+-      apiUrl: `${middlewareUrl}/commerce`,
+-      requestOptions: { headers: getRequestHeaders },
++  ({ buildModule, middlewareModule, middlewareUrl, getRequestHeaders }) => ({
++      commerce: buildModule(middlewareModule<Endpoints>, {
```
Comment (jagoral) [inline packages/storefront/packages/nuxt/CHANGELOG.md:17]: UnifiedEndpoints

File: packages/storefront/packages/nuxt/CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,45 @@
...
+- **[CHANGED]** deprecate `getCookieHeader`, use `getRequestHeaders` instead
+
+```diff [sdk.config.ts]
+- import { UnifiedApiExtension } from "storefront-middleware/types"
++ import { UnifiedEndpoints } from "storefront-middleware/types"
+
+export default defineSdkConfig(
+-  ({ buildModule, middlewareModule, middlewareUrl, getCookieHeader }) => ({
++  ({ buildModule, middlewareModule, middlewareUrl, getRequestHeaders }) => ({
+      commerce: buildModule(middlewareModule<Endpoints>, {
```
Comment (jagoral) [inline packages/storefront/packages/nuxt/CHANGELOG.md:34]: UnifiedEndpoints

Comment (WojtekTheWebDev) [reply packages/storefront/packages/nuxt/CHANGELOG.md]: fixed

File: packages/storefront/packages/nuxt/CHANGELOG.md (hunk 7 above)
Comment (WojtekTheWebDev) [reply packages/storefront/packages/nuxt/CHANGELOG.md]: fixed

**Review by jagoral — APPROVED** [review]

**Pull request merged**